### PR TITLE
13519 Added validation for manufacturer name

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity###

*Description of changes:*
- Added form validation for manufacturer's name

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue [hunk 1]
```diff
@@ -219,6 +227,13 @@ export default defineComponent({
       )
     )
 
+    const manufacturerNameRules = computed((): Array<Function> =>
+      customRules(
+        required("Enter a manufacturer's name or enter \"Unknown\" "),
+        maxLength(65)
+      )
+    )
+
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue:236]: I don't mind the shorthand approach for function expressions (ie arrow function with no parameters{})!
Just keep in mind, most other components include computed values in the `localState` as well as parameters and explicit return statements.
ie `someValue: computed((): boolean => { return true })`

File: ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue [hunk 2]
```diff
@@ -261,13 +276,15 @@ export default defineComponent({
...
       hasError,
+      nameRef,
       yearRef,
       makeRef,
       modelRef,
       makeModelComboForm,
       manufactureYearRules,
       makeRules,
       modelRules,
+      manufacturerNameRules,
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue:287]: Just to add to the above, by including computed values in the `localState` you don't need to explicitly return them here, as there are included in the spread of the `localState` 
line 290 `...toRefs(localState)`

Just some info, no action required :)

**Review by cameron-eyds — APPROVED** [review]
Just looking at the UI code here, looks good 👍 
Just a few comments, no action required!

File: ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue [hunk 3]
```diff
@@ -219,6 +227,13 @@ export default defineComponent({
       )
     )
 
+    const manufacturerNameRules = computed((): Array<Function> =>
+      customRules(
+        required("Enter a manufacturer's name or enter \"Unknown\" "),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue:232]: Design calls for this error message? 
`Enter a manufacturer's name or enter "Unknown"`

Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue:232]: Yep, that is what I saw in the ticket (final bullet point)

**Solution:

-remove the word "(Optional)" from the field
-validate as a required field - triggered by user going to review and confirm step
-show validations (red bar/red side label) in Step 1 and show step not completed on Review and Confirm screen
-stepper green check mark icon should not appear on Step 1 unless all required info has been added
-show required field validation error message: Enter a manufacturer's name or enter "Unknown"**

**Review by ChasBoucher — APPROVED** [review]
Looks Good.

File: ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue:232]: Sounds good, thanks for verifying!

**Pull request merged**